ElrondNetwork/elrond-go
# Fallback consensus through lower num of signatures

Acceptance of a temporary lower threshold for number of signatures needed in metachain consensus (ex. 1/2 + 1 (~51%) instead 2/3 + 1 (~67%) ) will be done only if:

1) there is a stuck situation (> 50 rounds without a new block created)

2) block which can not be created is the start of epoch meta block

**How to be tested ?**

1) Start a normal testnet with 7 nodes in each shard. In metachain consensus will be achieved if there are 2/3+1 (67%) signatures collected from the whole consensus group which is also 7 (equal with number of eligible nodes). So we need at least 5 signatures for each proposed block.
2) After a while, just before an epoch to be changed, we should close 3 nodes from metachain just to be sure that the consensus will not be achieved anymore. With only 4 nodes nothing new will be produced. The tricky part will come after 50 rounds without any new block produced and when the time for a new epoch will be reached. In this situation the fallback mechanism will be activated and the new threshold of 1/2 + 1 (51%) will be applied. This means that 4 signatures from the total of 7, will be enough now for a proposed block to be accepted. (only if the proposed block will be start of epoch meta block and there are more than 50 rounds without a new block created)
3) We can test then a fast replay for the whole blockchain to check also if the start of epoch meta blocks with lower threshold of signatures will be accepted by the nodes which are syncing, either if they are shard nodes or meta nodes.
